[17.0][MIG] sale_automatic_workflow_periodicity

henrybackman opened the pull request:
Migration from 14.0 to 17.0

TDu committed:
Add sale_automatic_workflow_frequency

This module allows to set a frequency at which a workflow is going to be executed.

The frequency of execution can be set in seconds in the workflow configuration. The
next execution time is displayed below it.

Ivorra78 committed:
Added translation using Weblate (Spanish)

henrybackman committed:
[IMP] sale_automatic_workflow_periodicity: pre-commit auto fixes

rousseldenis commented:
/ocabot migration sale_automatic_workflow_periodicity

rousseldenis commented on sale_automatic_workflow/models/automatic_workflow_job.py:
```diff
@@ -176,9 +176,12 @@ def _register_payment_invoice(self, invoice):
     def _handle_pickings(self, sale_workflow):
         pass
 
+    def _sale_workflow_domain(self, workflow):
```
@henrybackman Please put these changes in `sale_automatic__workflow`in another PR (moreover those changes should not be in same commit as migration of the other module)

henrybackman replied:
Thank you! I've now separated the commits

henrybackman committed:
[MIG] sale_automatic_workflow_periodicity: Migration to 17.0

henrybackman committed:
[FIX] sale_automatic_workflow: add domain methods to override

When migrating the sale_automatic_workflow_periodicity to 17.0 needed
these to make the module work

alexey-pelykh reviewed (changes requested):
Thanks for picking up this migration.

A few things to address:

**View XML -- duplicate `name` attribute**

The inherited view inserts a `<div class="container" name="sale_order_confirm">` before the existing `<div ... name="sale_order_confirm">`. Having two elements with the same `name` in the form will break xpath resolution for any other module targeting `//div[@name='sale_order_confirm']`. The original 14.0 version used `<group name="workflow_periodicity_exec">` with its own unique name -- please use a distinct name here as well (e.g. `name="workflow_periodicity_exec"`).

Also, the 14.0 view used `<group>` elements which provide automatic label alignment. The 17.0 version switches to manual `<div class="row">` / `<label>` / `<span>` / `<field>` layout to match the parent form style, which is fine, but the `next_execution` field lost its `readonly` attribute. Since this field is computed by the cron and shouldn't be edited manually, consider adding `readonly="1"` on the field in the view (or on the model definition as in 14.0).

**Translation files reference Odoo 14.0**

Both `i18n/sale_automatic_workflow_periodicity.pot` and `i18n/es.po` have `Project-Id-Version: Odoo Server 14.0`. These should be regenerated for 17.0 (the .pot at minimum -- `oca-gen-addon-readme` / the export tool will take care of it).

**`sale_automatic_workflow` changes scope**

As @rousseldenis already noted, the hook-point additions to `sale_automatic_workflow` (the new `_sale_workflow_domain` and `_workflow_process_to_run_domain` methods) should be submitted as a separate PR against `sale_automatic_workflow`. This PR should only contain the new module. That prerequisite PR needs to land first, and this migration PR should be rebased on top of it.

rousseldenis commented:
Closing this as very old. Feel free to reopen it if needed
